### PR TITLE
Add new lookup plugin json_template

### PR DESCRIPTION
**Pull request by trishnaguha**
Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>
Fixes https://github.com/ansible-network/network-engine/issues/32
Tests
Playbook:
```
- name: generate config in JSON
  debug: msg="{{ lookup('json_template', 'config.json') }}"
```

JSON file:
```
{
    "object": [
	{
	    "object": [
		{
		    "value": "GigabitEthernet0/0",
		    "key": "name"
		},
		{
		    "value": "iGbE",
		    "key": "type"
		}, 
		{
		    "value": "1500",
		    "key": "mtu"
		}, 
		{
		    "value": "Configured by Ansible",
		    "key": "description"
		}
	    ],
	    "key": "config"
        }
    ],
    "key": "GigabitEthernet0/0"
}
````

Return
```
ok: [localhost] => {
    "msg": {
        "GigabitEthernet0/0": {
            "config": {
                "description": "Configured by Ansible", 
                "mtu": 1500, 
                "name": "GigabitEthernet0/0", 
                "type": "iGbE"
            }
        }
    }
}
```

### CODE REVIEW

File: lookup_plugins/json_template.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+"""
+
+import json
+
+from ansible.errors import AnsibleError, AnsibleParserError
+from ansible.plugins.lookup import LookupBase
+from ansible.module_utils._text import to_bytes
+from ansible.module_utils.network.common.utils import to_list
+
+try:
```
Comment (privateip) [inline lookup_plugins/json_template.py]: You can import display from `ansible.plugins.lookup`

File: lookup_plugins/json_template.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    from ansible.utils.display import Display
+    display = Display()
+
+
+class LookupModule(LookupBase):
+
+    def run(self, terms, variables, **kwargs):
+        ret = list()
+
+        for term in to_list(terms[0]):
```
Comment (privateip) [inline lookup_plugins/json_template.py]: This should only use a single path not a list of paths.  Change `to_list(terms[0])` to `terms[0]` and remove the `for` loop

File: lookup_plugins/json_template.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            display.debug("File lookup term: %s" % term)
+
+            lookupfile = self.find_file_in_search_path(variables, 'files', term)
+            display.vvvv("File lookup using %s as file" % lookupfile)
+            try:
+                if lookupfile:
+                    with open(to_bytes(lookupfile, errors='surrogate_or_strict'), 'rb') as f:
+                        ret.append(json.load(f))
```
Comment (privateip) [inline lookup_plugins/json_template.py]: This needs to pass through `network_engine.plugins.template.json_template`

**Review by privateip — CHANGES_REQUESTED** [review]

**Review by privateip — APPROVED** [review]

**Comment by trishnaguha** [comment]
This needs to be cherry-picked to devel.

**Pull request merged**